src/main/java/bio/terra/service/load/LoadDao.java

```diff
@@ -292,47 +290,34 @@ public BulkLoadResultModel makeBulkLoadResult(UUID loadId) {
     return jdbcTemplate.query(
         bulkLoadResultSql,
         params,
-        (ResultSetExtractor<BulkLoadResultModel>)
-            rs -> {
-              BulkLoadResultModel result =
-                  new BulkLoadResultModel()
-                      .succeededFiles(0)
-                      .failedFiles(0)
-                      .notTriedFiles(0)
-                      .totalFiles(0);
-
-              while (rs.next()) {
-                BulkLoadFileState state = BulkLoadFileState.fromValue(rs.getString("state"));
-                if (state == null) {
-                  throw new CorruptMetadataException("Invalid file state");
-                }
-                switch (state) {
-                  case RUNNING:
-                    logger.info("Unexpected running loads: " + rs.getInt("statecount"));
-                    throw new CorruptMetadataException("No loads should be running!");
-
-                  case FAILED:
-                    result.setFailedFiles(rs.getInt("statecount"));
-                    break;
-
-                  case NOT_TRIED:
-                    result.setNotTriedFiles(rs.getInt("statecount"));
-                    break;
-
-                  case SUCCEEDED:
-                    result.setSucceededFiles(rs.getInt("statecount"));
-                    break;
-
-                  default:
-                    throw new CorruptMetadataException("Invalid load state");
-                }
-                result.setTotalFiles(
-                    result.getFailedFiles()
-                        + result.getNotTriedFiles()
-                        + result.getSucceededFiles());
+        rs -> {
+          BulkLoadResultModel result =
+              new BulkLoadResultModel()
+                  .succeededFiles(0)
+                  .failedFiles(0)
+                  .notTriedFiles(0)
+                  .totalFiles(0);
+
+          while (rs.next()) {
+            BulkLoadFileState state = BulkLoadFileState.fromValue(rs.getString("state"));
+            if (state == null) {
+              throw new CorruptMetadataException("Invalid file state");
+            }
+            switch (state) {
+              case RUNNING -> {
+                logger.info("Unexpected running loads: " + rs.getInt("statecount"));
+                throw new CorruptMetadataException("No loads should be running!");
               }
-              return result;
-            });
+              case FAILED -> result.setFailedFiles(rs.getInt("statecount"));
```

Comment:
I would go one step further here and take advantage of the exhaustiveness feature of switch expressions. The main benefit there is that `default` is no longer needed, and if the enum is ever changed, the code won't compile until the `switch` is changed. The only requirement is that each case must return a value, which pretty naturally maps to an `IntConsumer` here. So one way to write that is:
```
            final int statecount = rs.getInt("statecount");
            IntConsumer useCount =
                switch (state) {
                  case RUNNING -> {
                    logger.info("Unexpected running loads: {}", statecount);
                    throw new CorruptMetadataException("No loads should be running!");
                  }
                  case FAILED -> result::setFailedFiles;
                  case NOT_TRIED -> result::setNotTriedFiles;
                  case SUCCEEDED -> result::setSucceededFiles;
                };
            useCount.accept(statecount);
```

Reply:
Hm, this was funny, when I tried this I commented out one of the cases to make sure the code wouldn't compile (since that would make the switch non-exhaustive).  I was surprised to find that it compiled:

```
(base) okotsopo@wm111-e35 jade-data-repo % ./gradlew compileJava
 …
BUILD SUCCESSFUL in 5s
```

I did like breaking out `statecount` for reuse, but I found the IntConsumer made the code harder to read so I left it out.

Reply:
> Hm, this was funny, when I tried this I commented out one of the cases to make sure the code wouldn't compile (since that would make the switch non-exhaustive). I was surprised to find that it compiled:

Did you try this using the `IntConsumer` form? The exhaustiveness only works if you use the `switch` as an expression. If you use the new-style switch as a statement, then it doesn't enforce it and you still need `default` to catch missing cases.

When I try this, it's flagged as an error in IntelliJ, and compiling produces this error:
```
❯ ./gradlew compileJava
/Users/pshapiro/source/jade-data-repo/src/main/java/bio/terra/service/load/LoadDao.java:309: error: the switch expression does not cover all possible input values
                switch (state) {
                ^
```

Reply:
Got it, thanks.  I did not use the `IntConsumer` form, so that explains it.